app/controllers/course/assessment/submissions_controller.rb

```diff
@@ -24,6 +24,14 @@ def edit
   end
 
   def update
+    if params[:submit_answer]
```

Comment:
You'd have to also consider what happens if the student does this on a submission that should not be automatically graded (later, get the logic working first.)

I believe that just grading the answer will work because there is a link for each answer to an auto_grading record. That tells you which job it is running as. The answer should automatically transition to the submitted state (hidden field with workflow_state value as submitted?) then the grading will kick itself off.

If it is a guided assessment, then redirect to the answer with a grading job?